payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayPaymentMethodLauncherViewModel.kt

```diff
@@ -123,68 +127,106 @@ internal class GooglePayPaymentMethodLauncherViewModel(
     internal class Factory(
         private val application: Application,
         private val args: GooglePayPaymentMethodLauncherContract.Args,
-        private val enableLogging: Boolean = false,
-        private val workContext: CoroutineContext = Dispatchers.IO
-    ) : ViewModelProvider.Factory {
+        owner: SavedStateRegistryOwner,
+        defaultArgs: Bundle? = null
+    ) : AbstractSavedStateViewModelFactory(owner, defaultArgs),
+        Injectable<Factory.FallbackInjectionParams> {
+
+        internal data class FallbackInjectionParams(
+            val application: Application,
+            val enableLogging: Boolean,
+            val publishableKey: String,
+            val stripeAccountId: String?,
+            val productUsage: Set<String>
+        )
+
+        @Inject
```

Comment:
These could be simplified with a subcomponent, please refer to [PaymentSheetViewModel.Factory](https://github.com/stripe/stripe-android/blob/c2330a129906a8beb9da0379ca42aae9dcdb9a56/paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt#L443) an example

Reply:
Yes, I'll get there, this is WIP.